pkg/kubelet/container/runtime.go

```diff
@@ -118,9 +118,9 @@ type Runtime interface {
 	// GetRuntimeConfigInfo returns runtime's configuration details, eg: if user-namespaces are enabled or not
 	GetRuntimeConfigInfo() (*RuntimeConfigInfo, error)
 	// GetHostUID returns the uid from the host usernamespace that is mapped to the container usernamespace uid, containerUID
-	GetHostUID(containerUID int) (int, error)
+	GetHostUID(containerUID uint32) (uint32, error)
```

Comment:
Not sure if `uid_t` and `gid_t` C types changes on different architectures. Have you checked? 

From a quick look, seems to be [unsigned int in gnu libc C][link]. I think an unsigned int can vary it's size between architectures and maybe not always be `uint32`. Maybe worth checking or double check map between go types and C types? Maybe there is one that matches unsigned int and that may be a better choice? My guess is that `uint` might be that, but haven't checked :D


[link]: https://www.gnu.org/software/libc/manual/html_node/Reading-Persona.html

Reply:
That's a good question. In golang the os package uses int, however in the `Stat_t` it's `uint32` https://golang.org/pkg/syscall/#Stat_t. It seems to me that it's not that clear what to use in golang https://github.com/golang/go/issues/8537, https://github.com/golang/go/issues/6495.
I'm fine with switching to uint.

Reply:
Ohh, makes sense to keep it like this, then! :)